### ISSUE TITLE
[BUG] Merge on empty dataframe fails

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#690]
Merging on an empty dataframe fails. 

For context dask.dataframe does this in order to get dtypes and column names without doing much work.

```python
import cudf
df = cudf.DataFrame({'x': []})
df.merge(df, on=['x'])
```

```python-traceback
ERROR: CUDA Runtime call cudaPeekAtLastError() in line 566 of file /home/nfs/mrocklin/cudf/cpp/src/join/joining.cu failed with invalid device pointer (17).
---------------------------------------------------------------------------
GDFError                                  Traceback (most recent call last)
<ipython-input-3-a5f3db0f3305> in <module>
----> 1 df.merge(df, on=['x'])

~/cudf/python/cudf/dataframe/dataframe.py in merge(self, other, on, how, lsuffix, rsuffix, type, method)
   1050
   1051         cols, valids = cpp_join.join(lhs._cols, rhs._cols, on, how,
-> 1052                                      method=method)
   1053
   1054         df = DataFrame()

~/cudf/python/cudf/bindings/join.pyx in cudf.bindings.join.join()

~/cudf/python/cudf/bindings/join.pyx in cudf.bindings.join.join()

~/cudf/python/cudf/bindings/cudf_cpp.pyx in cudf.bindings.cudf_cpp.check_gdf_error()

GDFError: CUDA ERROR. b'cudaErrorInvalidDevicePointer': b'invalid device pointer'
```

**Comment by mrocklin** [comment]
cc @randerzander

**Comment by kkraus14** [comment]
cc @jrhemstad @kaatish should be easier to triage this

**Comment by jrhemstad** [comment]
Aha, I suspected that this was the cause of https://github.com/rapidsai/dask-cudf/issues/66

**Comment by jrhemstad** [comment]
@mrocklin or @kkraus14 could you open a PR that adds a pytest to reproduce this so that I have something to use for debugging?

**Comment by kkraus14** [comment]
> @mrocklin or @kkraus14 could you open a PR that adds a pytest to reproduce this so that I have something to use for debugging?

Yessir. Give me a few.

**Comment by kkraus14** [comment]
Fixed in #691
